plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -258,9 +263,11 @@ public void testSelectivelyOptimizingLeavesEqualityDeletes()
         assertUpdate("CREATE TABLE " + tableName + " WITH (partitioning = ARRAY['nationkey']) AS SELECT * FROM tpch.tiny.nation", 25);
         Table icebergTable = loadTable(tableName);
         writeEqualityDeleteToNationTable(icebergTable, Optional.of(icebergTable.spec()), Optional.of(new PartitionData(new Long[]{1L})));
+        assertQuery("SELECT count(*) FROM \"" + tableName + "$files\" WHERE content = 2", "VALUES 1");
         query("ALTER TABLE " + tableName + " EXECUTE OPTIMIZE WHERE nationkey < 5");
         assertQuery("SELECT * FROM " + tableName, "SELECT * FROM nation WHERE regionkey != 1 OR nationkey != 1");
         Assertions.assertThat(loadTable(tableName).currentSnapshot().summary().get("total-equality-deletes")).isEqualTo("1");
+        assertQuery("SELECT count(*) FROM \"" + tableName + "$files\" WHERE content = 2", "VALUES 0");
```

Comment:
Note to @krvikash. After Optimize `currentSnapshot` has `total-equality-deletes` is 1, but `$files` return 0 equality delete files.